src/client/clientmap.cpp

```diff
@@ -919,23 +913,13 @@ void ClientMap::updateDrawListShadow(const v3f &shadow_light_pos, const v3f &sha
 				continue;
 			}
 
-			float range = shadow_range;
-
-			float d = 0.0;
-			if (!isBlockInSight(block->getPos(), camera_position,
-					    camera_direction, camera_fov, range, &d))
+			v3f block_pos = intToFloat(block->getPos() * MAP_BLOCKSIZE, BS);
+			v3f projection = shadow_light_pos + shadow_light_dir * shadow_light_dir.dotProduct(block_pos - shadow_light_pos);
+			if (projection.getDistanceFrom(block_pos) > radius)
 				continue;
 
 			blocks_in_range_with_mesh++;
 
-			/*
-				Occlusion culling
-			*/
-			if (isBlockOccluded(block, cam_pos_nodes)) {
```

Comment:
Why is this no longer needed? Did culling blocks from the shadow view cause glitches?

Reply:
@lhofhansl yes, occlusion culling is causing problems because mapblock shadows are rendered using backfaces. Also the frustum boundaries were incorrect for orthogonal projection.